src/wallet/scriptpubkeyman.cpp

```diff
@@ -304,6 +344,19 @@ void LegacyScriptPubKeyMan::MarkUnusedAddresses(const CScript& script)
                 WalletLogPrintf("%s: Topping up keypool failed (locked wallet)\n", __func__);
             }
         }
+
+        auto it = mapKeyMetadata.find(keyid);
+        if (it != mapKeyMetadata.end()){
+            CKeyMetadata meta = it->second;
+            if (!meta.hd_seed_id.IsNull() && meta.hd_seed_id != m_hd_chain.seed_id) {
+                bool internal = meta.key_origin.path[1] & ~BIP32_HARDENED_KEY_LIMIT != 0;
```

Comment:
This line is causing a -Wparentheses build warning `suggest parentheses around comparison in operand of ‘&’`

Reply:
Done